test-data/unit/fine-grained-attr.test

```diff
@@ -46,3 +46,32 @@ A.__attrs_attrs__.b
 
 [out]
 ==
+
+[case magicAttributeConsistency2]
+import m
+
+[file c.py]
+import attr
+
+@attr.s
+class Entry:
+    var: int
+[builtins fixtures/attr.pyi]
+
+[file m.py]
+import attr
+from c import Entry
+
+def func(e: Entry) -> None:
+    attr.asdict(e)
+
+[file m.py.2]
+import attr
+from c import Entry
+
+def func(e: Entry) -> int:
+    attr.asdict(e)
+    return 2
```

Comment:
The fix does work when tested locally. However, I couldn't reproduce it with this test case although I though I should be able to. Maybe the cache isn't serialized for it?

Open to ideas how to fix the test case.

Reply:
I was able to reproduce with `types-attrs`. Maybe try revendoring the attr fixture from typeshed from before https://github.com/python/typeshed/pull/5585 ?

Reply:
I believe we already use a custom stub for the test from `test-data/unit/lib-stub/attr`. That shouldn't be the issue, I think.

Rather the problem only happens if `c.py` is deserialized from cache and I'm not sure that is happing here. Might be that just both `c.py` and `m.py.2` get analyzed in the second run in which case the error won't show up.

Reply:
@hauntsaninja Got it working 🎉 Moving `AttrsInstance` from `lib-stub` to the test file did the job. Without the change in `plugins/attrs.py`, the test case is now failing.